### PR TITLE
Enable test reporting

### PR DESCRIPTION
**Pull request by devpow112**
Generate and submit a D2L format test report so it appears in https://test-reporting.d2l.dev dashboards.

### CODE REVIEW

File: d2l-test-reporting.config.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "tool": "n/a",
+  "experience": "n/a",
```
Comment (devpow112) [inline d2l-test-reporting.config.json:3]: I couldn't find anything in the product taxonomy that made sense for these here. Any ideas @sarahboyd?

Comment (dlockhart) [reply d2l-test-reporting.config.json:3]: That's a pretty common problem for us haha.

Comment (sarahboyd) [reply d2l-test-reporting.config.json:3]: I'd think UI Platform & Components for tool, under Software Services  - would that make sense?
image

File: d2l-test-reporting.config.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  "overrides": [
```
[the remark this answers is not in the document]
Comment (devpow112) [reply d2l-test-reporting.config.json:4]: Probably. This is just the format we have so far. I'm hoping to make improvements. I think for the `BrightspaceUI/testing` framework we can put in some default eventually. Right now they are required in the config file to work based on how the test reporting stuff is coded.

File: d2l-test-reporting.config.json (hunk 1 above)
Comment (devpow112) [reply d2l-test-reporting.config.json:3]: Ah yes. Didn't see that. That diagram makes me go blind. I'm glad your sight isn't lost on it

Comment (sarahboyd) [reply d2l-test-reporting.config.json:3]: I've just spent too much time looking through it 😆 🙃 🙈

File: d2l-test-reporting.config.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (devpow112) [reply d2l-test-reporting.config.json:4]: I think eventually I can make them code first options in the reporter so we can send them in when we add the reporter. Then we could infer some stuff based on the group names the testing stuff does.

[the remark this answers is not in the document]
Comment (devpow112) [reply d2l-test-reporting.config.json:4]: Or maybe let there be a generator function? Either way I think if we get in some more use cases I can figure out something we can use for all of them.

[the remark this answers is not in the document]
Comment (dlockhart) [reply d2l-test-reporting.config.json:4]: Sounds good!

File: d2l-test-runner.config.js [hunk 3]
```diff
@@ -1,3 +1,6 @@
+import { env } from 'node:process';
+
 export default {
-	pattern: type => `**/test/*.${type}.js`
+	pattern: type => `**/test/*.${type}.js`,
+	testReporting: !!env['CI']
```
Comment (devpow112) [inline d2l-test-runner.config.js:5]: This gets vdiff as well for now. since it's run from within a GitHub Action I didn't want to change the action to enable test reporting. This can go away/become the new default eventually so `BrightspaceUI/testing` always generates the reports when run in CI.

File: helpers/test/dismissible.test.js [hunk 4]
```diff
@@ -66,7 +66,7 @@ describe('dismissible', () => {
 	});
 
 	[undefined, null, 0, 'hello', []].forEach((cb) => {
-		it(`should handle invalid callback: ${cb}`, () => {
+		it(`should handle invalid callback: ${JSON.stringify(cb)}`, () => {
```
Comment (devpow112) [inline helpers/test/dismissible.test.js:69]: Currently test reporting doesn't allow leading or trailing spaces so we can have the name be consistent. We've had a few come up were the name was the same minus spaces. There are future changes that will probably make this a non-issue but for now changing to `stringfy` the `cb` for the test name since `[]` default serialize to just nothing meaning the test name is `should handle invalid callback: ` with a trailing space.

File: components/form/test/form-element.test.js [hunk 5]
```diff
@@ -72,7 +72,7 @@ describe('form-element', () => {
 			{ forceInvalid: false, validationError: null },
 		].forEach(({ forceInvalid, validationError }) => {
 
-			it('should be invalid if force validate is true or there is a validation error ', async() => {
+			it('should be invalid if force validate is true or there is a validation error', async() => {
```
Comment (devpow112) [inline components/form/test/form-element.test.js:75]: Same here

File: components/tooltip/test/tooltip.test.js [hunk 6]
```diff
@@ -274,7 +274,7 @@ describe('d2l-tooltip', () => {
 			expect(tooltip.showing).to.be.true;
 		});
 
-		it('should not hide when ESC key is pressed ', async() => {
+		it('should not hide when ESC key is pressed', async() => {
```
Comment (devpow112) [inline components/tooltip/test/tooltip.test.js:277]: Same here

**Comment by devpow112** [comment]
Doing this integration let me find a few edge cases in things. Have to resolve 1 kore issue and then this should be good to go. Data is showing up as expected
* [https://test-reporting.d2l.dev/metrics](https://test-reporting.d2l.dev/metrics?var-githubOrganizations=BrightspaceUI&var-githubRepositories=core&var-branchType=All&from=now-24h&to=now)
* [https://test-reporting.d2l.dev/drill-down](https://test-reporting.d2l.dev/drill-down?var-githubOrganizations=BrightspaceUI&var-githubRepositories=core&var-branchType=All&from=now-24h&to=now) (ignore the test history ones in this, they aren't filtering properly)

**Comment by devpow112** [comment]
I think this is a good start. It will add data for all CI runs. It might be nice to eventually have test run on release so we can track the tests for the released version instead of the random PR's people put up but this is handy for now.

**Comment by devpow112** [comment]
Working on fixing the double summary issue but not a deal breaker for now.

**Comment by devpow112** [comment]
Won't be merging this today so take your time. Not a fan of merging stuff that could be risky (not sure this is really risky) before the weekend.

**Review by dlockhart — APPROVED** [review]
Awesome, thanks for setting this up!

**Review by sarahboyd — APPROVED** [review]
Yay! this is exciting 🎉
